[Tariq-droid] iP

Tariq-droid opened the pull request:
CS2103 Week 3 PR

Commit:
Add Gradle support

Tariq-droid committed:
testing

Tariq-droid committed:
Level-1

Tariq-droid committed:
Level-2

Tariq-droid committed:
Level-3

Tariq-droid committed:
Level-4

Tariq-droid committed:
A-TextUiTesting

Tariq-droid committed:
Level-5

Tariq-droid committed:
Level-6

Tariq-droid committed:
branch-level-7

Tariq-droid committed:
branch-level-8

Tariq-droid committed:
level-8

Tariq-droid committed:
A-MoreOOP

Tariq-droid committed:
A-Packages
Moved all to package duke

Tariq-droid committed:
Merge remote-tracking branch 'origin/add-gradle-support'

Tariq-droid committed:
Added gradle to the project

Tariq-droid committed:
Added Jar file

Tariq-droid committed:
Added test cases for Parser and Duke with Junit support

Tariq-droid committed:
Added header comments

Tariq-droid committed:
Adhered to the coding standard

Tariq-droid committed:
Added find function

Tariq-droid committed:
no message

Tariq-droid committed:
no message

Tariq-droid committed:
Merge tag 'Level-9'

Tariq-droid committed:
Merge remote-tracking branch 'origin/heads/A-CodingStandard'

Tariq-droid committed:
Merge branch 'branch-A-JavaDoc'

Tariq-droid committed:
Merge branch 'branch-A-CodingStandard'

Tariq-droid committed:
Update README.md

Tariq-droid committed:
Update README.md

shirsho-12 commented on src/main/java/seedu/duke/Parser.java:
```diff
@@ -0,0 +1,56 @@
+
+    /**
+     * Takes in the command and chooses what function should be used.
+     *
```
To my knowledge, this extra space is a JavaDoc violation. It is troublesome as VSCode automatically does it for you. Would suggest setting up the checkstyles soon.

shirsho-12 commented on src/main/java/seedu/duke/Storage.java:
```diff
@@ -0,0 +1,170 @@
+
+import java.util.ArrayList;
+
+public class Storage {
```
Would suggest adding JavaDocs to the class and its functions.

shirsho-12 commented on src/main/java/seedu/duke/TaskList.java:
```diff
@@ -0,0 +1,194 @@
+    public void addDeadline(String[] echoSplit, int print){
+        String task = "";
+        String date = "";
+
+        for (int i = 1; i < echoSplit.length; i++) {
+            if(echoSplit[i].equals("/by") || echoSplit[i].equals("by:")){
+
+                for (int j = 1; j < i; j++) {
+                    if(j == i-1)
+                        task += echoSplit[j];
+                    else
+                        task += echoSplit[j] + " ";
+                }
+                for (int j = i + 1; j < echoSplit.length; j++) {
+                    if(j == echoSplit.length - 1)
+                        date += echoSplit[j];
+                    else
+                        date += echoSplit[j] + " ";
+                }
+
+                list.add(new Deadline(task, date));
+                if(print == 0) {
+                    System.out.println(echoSplit[i+1]);
+                    System.out.println("    -------------------------------------------\n    "
```
Nice output handling.

Tariq-droid replied:
Thanks

shirsho-12 reviewed (commented):
I generally like the codebase, but certain classes do not have JavaDocs yet. I'd also suggest further encapsulation of UI features into the UI class.

Tariq-droid committed:
Added Level-10

Tariq-droid committed:
readme

Tariq-droid committed:
Merge branch 'master' of https://github.com/Tariq-droid/ip

# Conflicts:
#	README.md

Tariq-droid committed:
just checking

Jxleejiaxin commented:
You could look at the example provided [addressbook-level2](https://github.com/se-edu/addressbook-level2) to abstract out printing of output to UI class like @shirsho-12 mentioned. Otherwise, LGTM. Keep up the good work!

Tariq-droid committed:
no message

Tariq-droid committed:
Added assertions method

Added asome assertions in the code to make sure no problems happen

These modifications are in another branch and will merge it with the master later

Tariq-droid committed:
Fixed the code quality in some parts

Mostly worked on the TaskList and Storage class

There were many nested if statements and a long method that were fixed

Tariq-droid committed:
Merge pull request #1 from Tariq-droid/branch-A-Assertions

Added assertions method

Tariq-droid committed:
Merge branch 'master' of https://github.com/Tariq-droid/ip

Tariq-droid committed:
Merge branch 'master' into branch-A-CodeQuality

Tariq-droid committed:
Merge pull request #2 from Tariq-droid/branch-A-CodeQuality

Fixed the code quality in some parts

Tariq-droid committed:
Merge branch 'master' of https://github.com/Tariq-droid/ip

Tariq-droid committed:
no message

Tariq-droid committed:
Added docs files

Tariq-droid committed:
updated user guide

Tariq-droid committed:
Releasing the beast

Tariq-droid committed:
Add User guide

Tariq-droid committed:
Add Ui.png via upload

Tariq-droid committed:
Update README.md

Tariq-droid committed:
Update README.md

Tariq-droid committed:
Update README.md

Tariq-droid committed:
Merge branch 'master' of https://github.com/Tariq-droid/ip

Tariq-droid committed:
Fixed the IP based on peer review
Added more abstraction
Added more tests
Fixed many bugs
and many more

Tariq-droid committed:
V1.0 Release

Tariq-droid committed:
Update README.md

Tariq-droid committed:
Code refinement WIP

Tariq-droid committed:
Add abstraction and fix bugs
Added abstraction to Storage, TaskList. Fixed bugs and edge cases

Tariq-droid committed:
Merge branch 'master' of https://github.com/Tariq-droid/ip
